File: pulse-kotlin/src/commonTest/kotlin/kotlin/com/fluxchess/pulse/kotlin/FileTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package kotlin.com.fluxchess.pulse.kotlin

import com.fluxchess.pulse.kotlin.files
import kotlin.test.Test
import kotlin.test.assertEquals

class FileTest {
	@Test
	fun `Files array should be valid`() {
		for (file in files.indices) {
			assertEquals(files[file], file)
		}
	}
}

File: pulse-go/internal/pulse/file_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

import (
	"testing"
)

func TestFile(t *testing.T) {
	t.Run("Files array should be valid", func(t *testing.T) {
		for file := range Files {
			if Files[file] != file {
				t.Errorf("wanted files at index %d to be %v, got %v", file, file, Files[file])
			}
		}
	})
}
